src/renderer/widgets/Background.cpp

```diff
@@ -97,5 +171,124 @@ bool CBackground::draw(const SRenderData& data) {
     texbox.round();
     g_pRenderer->renderTexture(texbox, *tex, data.opacity, 0, HYPRUTILS_TRANSFORM_FLIPPED_180);
 
-    return data.opacity < 1.0;
-}
\ No newline at end of file
+    return fade || data.opacity < 1.0; // actively render during fading
+}
+
+void CBackground::plantReloadTimer() {
+
+    if (reloadTime == 0) {
+        reloadTimer = g_pHyprlock->addTimer(std::chrono::hours(1), onReloadTimer, this, true);
+    } else if (reloadTime > 0)
+        reloadTimer = g_pHyprlock->addTimer(std::chrono::seconds(reloadTime), onReloadTimer, this, false);
+};
+
+void CBackground::onCrossFadeTimerUpdate() {
+
+    // Animation done: Unload previous asset, deinitialize the fade and pass the asset
+
+    if (fade) {
+        fade->crossFadeTimer.reset();
+        fade.reset(nullptr);
+    };
```

Comment:
; ?

Reply:
Nixlang has apperently corrupted my mind and now I put those everywhere. Sorry I’ll fix them up.